app/apollo/test/cluster.fga.spec.js

```diff
@@ -0,0 +1,568 @@
+      throw error;
+    }
+  });
+
+  // clusterbyClusterId without authorization
+  it('fgaUser01 does NOT have authorization to get cluster noAuthCluster by clusterID', async () => {
+    let response;
+    try {
+      const clusterId = 'noAuthCluster';
+      response = await clusterApi.byClusterID(fgaToken01, {
+        orgId: org01._id,
+        clusterId: clusterId,
+      });
+      expect(response.data.data.clusterByClusterId).to.equal(null);
+      expect(response.data.errors[0].message).to.be.a('string');
+      expect(response.data.errors[0].message).to.contain('You are not allowed');
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clusterByName
+  it('fgaUser01 has authorization to get cluster 1 by cluster name', async () => {
+    let response;
+    try {
+      const clusterId = testCluster1.cluster_id;
+      const clusterName = testCluster1.registration.name;
+      response = await clusterApi.byClusterName(fgaToken01, {
+        orgId: org01._id,
+        clusterName: clusterName,
+      });
+      const clusterByClusterName = response.data.data.clusterByName;
+      expect(clusterByClusterName.clusterId).to.equal(clusterId);
+      expect(clusterByClusterName.regState).to.equal('registering');  // record attr
+      expect(clusterByClusterName.status).to.equal('registered'); // created ~= updated
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clusterByName without authorization
+  it('fgaUser01 does NOT have authorization to get cluster noAuthCluster by cluster name', async () => {
+    let response;
+    try {
+      const clusterName = 'no-auth-cluster';
+      response = await clusterApi.byClusterName(fgaToken01, {
+        orgId: org01._id,
+        clusterName: clusterName,
+      });
+      expect(response.data.data.clusterByName).to.equal(null);
+      expect(response.data.errors[0].message).to.be.a('string');
+      expect(response.data.errors[0].message).to.contain('You are not allowed');
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clustersByOrgId
+  it('fgaUser01 has authorization to get all clusters by Org ID', async () => {
+    let response;
+    try {
+      response = await clusterApi.byOrgID(fgaToken01, {
+        orgId: org01._id,
+      });
+      const clustersByOrgId = response.data.data.clustersByOrgId;
+      expect(clustersByOrgId).to.be.an('array');
+      expect(clustersByOrgId).to.have.length(1);
```

Comment:
This test isn't completely useful because there _**is**_ only a single cluster.  Therefore it doesn't really verify it returns only authorized clusters.  Actually creating the `noAuthCluster`, as suggested earlier, will also address this shortcoming (though the test here should also verify that the single cluster returned is actually `testCluster1`)

Reply:
good for all fga files. they each have at least 2 of their respective types created

Reply:
Resolved